packages/core/src/components/overflow-list/overflowList.tsx

```diff
@@ -184,10 +216,15 @@ export class OverflowList<T> extends React.PureComponent<IOverflowListProps<T>,
                 }
                 const overflow = collapseFromStart ? [...state.overflow, next] : [next, ...state.overflow];
                 return {
+                    // set SHRINK mode unless a GROW is already in progress.
```

Comment:
why not always switch to shrink?

Reply:
for this condition https://github.com/palantir/blueprint/pull/2975/files#diff-8a0a6ef555cabf743f3603de6f56ec48R151.

Reply:
updated comment